### ISSUE TITLE
Development and Release Branches

### ISSUE THREAD

**Issue by FabrizioMoggio** [camaraproject/ReleaseManagement#10]
**Problem description**
in the Camara Versioning Guidelines document (https://github.com/camaraproject/Commonalities/blob/main/documentation/Camara_Versioning_Guidelines.md) it is described how to create Release Branches. It is not clear to me where the development is done (Release Branches or Main). I see two options and I'm not sure which is the one adopted by CAMARA.

Option 1:

- the **Main Branch** is where the External Developer can find the very last **released version** of a Family and, using the Release Tags, also the previous releases. 
- the **Release Branches** are meant for **development** of a new release. When the development is closed the code is merged in Main. Considering an API Family with more APIs inside, when all the APIs are merged into Main a new Release Tag, in Main, is added.

pro: 
-it is possible to develop in parallel different releases
-it is easier for the External Developer to find the very last version (Main Branch)

Option 2:

- the **Main Branch** is where the External Developer can find the work in progress. It is meant **for development**. When the work is consolidated a Release Tag is created and a Release Branch is created. Considering an API Family with more APIs inside, a Release Tag and a new Release Branch is done when all the APIs are completed.
- the Release Branches is where the External Developer can find a stable version. 

**Expected action**
Clarify this in the  Camara Versioning Guidelines document

**Comment by tanjadegroot** [comment]
The Release Mgmt team has defined the API Release Process here: 

The **Main Branch** is where the Internal Developers work on the API. 

- External Developers MUST NOT use the **"work-in-progress"** versions as they are unstable (between releases).
- External Developers SHALL only use public-release APIs for commercial deployments (made available at meta-release). **stable** public-release are recommended, as **initial** public release will still have a higher change rate. 
- External Developers MAY use pre-releases of APIs (alpha or release-candidate) AT THEIR OWN RISK. these are not recommended for commercial usage as internal to CAMARA.

The exception is for patches of public-release APIs, for which 
A **Maintenance Branch** SHALL only be created for updating a public-release API.
All other changes shall result in new API version development and release done on the Main Branch.

So the concept of "Release Branches" is deprecated and no longer used going forward.

The above information should allow to close this issue. 

Please review / comment the **extract** of this API Release Process for GitHub available in PR: https://github.com/camaraproject/ReleaseManagement/pull/31

**Comment by tanjadegroot** [comment]
@FabrizioMoggio Hi Fabrizio, are you OK to close this Issue ? thanks for you confirmation.

**Comment by tanjadegroot** [comment]
Finalized as described above and closed per lazy consensus as discussion in Release Management call on June 18th 2024
